### PR TITLE
Add ability to pass healthcheck setting for deployed apps

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#43

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java [hunk 1]
```diff
@@ -242,6 +245,15 @@ private int memory(AppDeploymentRequest request) {
 			request.getDeploymentProperties().getOrDefault(MEMORY_PROPERTY_KEY, valueOf(deploymentProperties.getMemory())));
 	}
 
+	private ApplicationHealthCheck healthCheck(AppDeploymentRequest request) {
+		ApplicationHealthCheck result = deploymentProperties.getHealthCheck();
+		String override = request.getDeploymentProperties().get(HEALTHCHECK_PROPERTY_KEY);
+		if (override != null) {
+			result = ApplicationHealthCheck.valueOf(override.toUpperCase());
```
Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: The doc will go in the server project. Waiting for https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/pull/172 to be merged (feel free to go ahead and merge it if you can), as it's likely to have conflicts.

Will provide a better error message here

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeploymentProperties.java [hunk 2]
```diff
@@ -37,6 +39,8 @@
 
 	public static final String SERVICES_PROPERTY_KEY = CLOUDFOUNDRY_PROPERTIES + ".services";
 
+	public static final String HEALTHCHECK_PROPERTY_KEY = CLOUDFOUNDRY_PROPERTIES + ".health-check";
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeploymentProperties.java]: The usage of the key `HEALTHCHECK_PROPERTY_KEY` doesn't include `stream` or `task` and being referred as `deployment` property from Deployment Request. Does this mean, when using as a stream deployment property it doesn't require `stream` or `task` in the prefix as in the App Deployer properties that are set via CF server startup?

I see all the other properties are handled the same way.

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeploymentProperties.java]: Correct. We're here in the *deployer* project. There is no concept of stream or tasks.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Rebased, squashed and merge as 6d40c3a

**Pull request closed without merging**